tests/disk/test_disk_exhaustion.py

```diff
@@ -0,0 +1,210 @@
+                               peer_ip_pc_pair]
+
+    # use first port of first peer_ip_ifaces pair as input port
+    # all ports in second peer_ip_ifaces pair will be output/forward port
+    ptf_port_idx = mg_facts["minigraph_ptf_indices"][peer_ip_ifaces_pair[0][1][0]]
+    # Some platforms do not support rif counter
+    try:
+        rif_counter_out = parse_rif_counters(
+            duthost.command("show interfaces counters rif")["stdout_lines"])
+        rif_iface = list(rif_counter_out.keys())[0]
+        rif_support = False if rif_counter_out[rif_iface]['rx_err'] == 'N/A' else True
+    except Exception as e:
+        logger.info("Show rif counters failed with exception: {}".format(repr(e)))
+        rif_support = False
+
+    pkt = testutils.simple_ip_packet(
+        eth_dst=duthost.facts["router_mac"],
+        eth_src=ptfadapter.dataplane.get_mac(0, ptf_port_idx),
+        ip_src=peer_ip_ifaces_pair[0][0],
+        ip_dst=peer_ip_ifaces_pair[1][0])
+
+    exp_pkt = pkt.copy()
+    exp_pkt.payload.ttl = pkt.payload.ttl - 1
+    exp_pkt = mask.Mask(exp_pkt)
+
+    exp_pkt.set_do_not_care_scapy(packet.Ether, 'dst')
+    exp_pkt.set_do_not_care_scapy(packet.Ether, 'src')
+
+    out_rif_ifaces, out_ifaces = parse_interfaces(
+        duthost.command("show ip route %s" % peer_ip_ifaces_pair[1][0])["stdout_lines"],
+        pc_ports_map)
+
+    out_ptf_indices = map(lambda iface: mg_facts["minigraph_ptf_indices"][iface], out_ifaces)
+
+    return pkt, ptf_port_idx, exp_pkt, out_ptf_indices, rif_support
+
+
+def test_disk_exhaustion(duthost, ptfadapter, tbinfo):
+    """Test SONiC basic performance(like ssh-connect, packet forward...) when disk is exhausted
+    Args:
+        duthost: DUT host object
+        ptfadapter: PTF adapter object
+        tbinfo: Testbed information
+    """
+
+    PKT_NUM = 1000
+    PKT_NUM_MIN = PKT_NUM * 0.9
+
+    # Construct packet and get params
+    pkt, ptf_port_idx, exp_pkt, out_ptf_indices, rif_support = construct_packet_and_get_params(duthost=duthost,
+                                                                                               ptfadapter=ptfadapter,
+                                                                                               tbinfo=tbinfo)
+
+    # Clear stats
+    duthost.command("portstat -c")
+
+    if rif_support:
+        duthost.command("sonic-clear rifcounters")
+    ptfadapter.dataplane.flush()
+
+    # Check SONiC image type and get default username and password to SSH connect
+    default_username_password = DEFAULT_SSH_CONNECT_PARAMS[get_image_type(duthost=duthost)]
+
+    # Simulate disk exhaustion and release space after 60 seconds
+    # Use command 'fallocate' to create large file, it's effective.
+    # Create a shell script to do the operations like fallocate and remove file,
+    # because when space is full, duthost.command() is not work
+    duthost.shell_cmds(cmds=[
+        "echo 'fallocate -l 50G /tmp/file' > /tmp/test.sh",
```

Comment:
* Although it is a temporary file, suggest not to name it like `file` which is too generic. How about name like `huge_dummy_file`.
* Why 50G? What if the tested platform has more free disk space than 50G?
* What is the limitation of single file size? Is 50G bigger than the limitation? Probably you need to use multiple big files if there is such limitation.
* Probably need to dynamically determine how many big dummy files need to generate based on free disk space?